[GLUTEN-5525][FOLLOWUP] Fix mvn versions:set does not work for shim submodules

JinHelin404 opened the pull request:
## What changes were proposed in this pull request?

mvn versions:set does not work for pom.xml in the shim/spark3x directory.

## How was this patch tested?

Run commend like follow and check pom version.

```
mvn -N --batch-mode versions:set -DnewVersion=${spark_version} versions:update-child-modules
```

And cases mentioned in [5519](https://github.com/apache/incubator-gluten/pull/5519) are also tested.

```
mvn clean
mvn clean install -DskipTests
mvn clean -Pbackends-velox -DskipTests
```

JinHelin404 commented:
@PHILO-HE @zhztheplayer Could you please take a look? Thanks!

PHILO-HE commented:
@JinHelin404, thanks for your pr!
Then, `sparkshim.module.name` is useless, can we remove it from property setting?

JinHelin404 commented:
@PHILO-HE Thanks for your reviewing! 

Sure we can. I removed the sparkshim.module.name property and added a default spark 34 profile in the pom of shims.

JinHelin404 commented on pom.xml:
```diff
@@ -309,7 +307,6 @@
       <properties>
         <sparkbundle.version>3.3</sparkbundle.version>
         <sparkshim.module.name>spark33</sparkshim.module.name>
-        <sparkshim.artifactId>spark-sql-columnar-shims-spark33</sparkshim.artifactId>
```
Fixed.

JinHelin404 commented:
@PHILO-HE Could you please take a look again? Thank you!

PHILO-HE reviewed (approved):
Thanks!
